src/metabase/api/database.clj

```diff
@@ -1042,6 +1043,18 @@
                                       [:in :collection_id (api/check-404 (seq (db/select-ids Collection :name schema)))])])
          (map api.table/card->virtual-table))))
 
+(api/defendpoint GET "/:id/models"
+  "Returns a list of models for a database."
+  [id]
+  {id su/IntGreaterThanZero}
+  (api/read-check Database id)
+  (let [cards          (-> (db/select Card, :database_id id, :archived false, :dataset true, {:order-by [[:%lower.name :asc]]})
+                           (hydrate :creator :collection))
```

Comment:
If you really need the creator and collection to be hydrated, then I think you should also have tests for it.

Otherwise, if someone accidentally removes it, FE will fail silently (assuming we don't have E2@ tests for it).

Reply:
Per [this conversation](https://metaboat.slack.com/archives/C505ZNNH4/p1664547245948629), I'm going to move this back to the api/card and test every column in this API. Unless the way the frontend changes caching entities, we should have full schema testing for the response of endpoints used by entities. And it's easier to manage this potential foot-gun with one endpoint, `api/card`.

Reply:
@qnkhuat After all that, I discovered I don't need a new endpoint at all. I've reused the `api/search` endpoint for this. So this PR needs no changes on the backend at all, except for correcting this docstring https://github.com/metabase/metabase/pull/25494/commits/24df5f6db6b31b8747ea36d63cc7e8584cf54f09

I appreciate your thorough review, so thank you anyhow.